generators/php/php_generator.ts

```diff
@@ -242,65 +251,62 @@ export class PhpGenerator extends CodeGenerator {
     }
     const nextBlock =
         block.nextConnection && block.nextConnection.targetBlock();
-    const nextCode = opt_thisOnly ? '' : this.blockToCode(nextBlock);
+    const nextCode = thisOnly ? '' : this.blockToCode(nextBlock);
     return commentCode + code + nextCode;
   };
 
   /**
-   * Gets a property and adjusts the value while taking into account indexing.
-   * @param {!Block} block The block.
-   * @param {string} atId The property ID of the element to get.
-   * @param {number=} opt_delta Value to add.
-   * @param {boolean=} opt_negate Whether to negate the value.
-   * @param {number=} opt_order The highest order acting on this value.
-   * @return {string|number}
+   * Generate code representing the specified value input, adjusted to take into
+   * account indexing (zero- or one-based) and optionally by a specified delta
+   * and/or by negation.
+   *
+   * @param block The block.
+   * @param atId The ID of the input block to get (and adjust) the value of.
+   * @param delta Value to add.
+   * @param negate Whether to negate the value.
+   * @param order The highest order acting on this value.
+   * @returns The adjusted value.
    */
-  getAdjusted(block, atId, opt_delta, opt_negate, opt_order) {
-    let delta = opt_delta || 0;
-    let order = opt_order || this.ORDER_NONE;
+  getAdjusted(block: Block, atId: string, delta = 0, negate = false, order = Order.NONE): string | number {
```

Comment:
In its current state, it doesn't look like this function ever returns a number. I guess this is overridable though?

Reply:
I started carrying over the previous type declaration (`@return {string | number}` from `javascript_generator.js` and did the same in all the others, but in fact only the `PythonGenerator`'s (similar code but differently named) `getAdjustedInt` ever returns a number.

I think the correct course of action here is to:

- Fix this PR to have `getAdjusted` be declared as returning only a `string`.
- In a separate PR, do the same for the other `getAdjusted` implementations.

Then, since PythonGenerator's gratuitously-different `getAdjustedInt` _does_ actually return numbers and some callers depend on this fact, fix this by:
- Introducing a new a string-only `getAdjusted` method.
- Updating callers to use it (and work correctly when it returns `'0'` instead of `0`).
- Marking `getAdjustedInt` as deprecated and removing it in due course.

I have added this to the list in #7600.

Reply:
(And yes: these changes would break any code that overrides `getAdjusted` with an implementation that returns numbers, if any such code exists and is implemented in TypeScript, but I do not think this egregiously worse than all the other random breakage we are likely to cause such developers just by dint issuing non-`any` generator typings.)